RedHatInsights/drift-frontend
# [DRFT-144] Add filters and sorting to url

To test:
- Create a comparison with any number of systems (to test best, add a system, baseline and hsp)
- See the filters and sort in the url
- Modify the url manually to achieve results
- Modify the filters and sorting via normal means (clicking buttons, typing in filters, etc)
- Add and remove systems to ensure that the url properly updates in response to your actions
- Ensure that the page updates in response to your edits to the url

For filtering and sorting, we are using the following rules based on the provided links.
- For filtering rules, see: https://jsonapi.org/recommendations/#filtering
- For sorting rules, see: https://jsonapi.org/format/#fetching-sorting

Here are some rules for how the urls are expected to look:
- Fact name filters are comma separated and follow `filter[name]=`
- State filters are comma separated and follow `filter[state]=`. If they are in the url, they should be selected. If they do not appear in the url, then they should be deselected
- Sorts are comma separated and follow `sort=`. A dash symbol in front of the sort signifies that it is a descending sort. No symbol signifies ascending sort. If fact is not added to the sort, the app will automatically set it to default descending. If state is not included in the sort, it will be set to no sort.

Here is an example of a complete filtering/sorting url:
`?filter[name]=bios,arch&filter[state]=same,incomplete_data&sort=-state,fact`

Here is an example of a full url with system ids and filtering/sorting:
`/insights/drift/?baseline_ids=<baseline_id>&hsp_ids=<hsp_id>&reference_id=<reference_id&system_ids=<system_id>&filter[state]=same,different,incomplete_data&sort=-state,fact`
